tests/js/snapshots.test.js

```diff
@@ -1,9 +1,168 @@
+  return await page.evaluate((selector) => {
+    const element = document.querySelector(selector);
+    const rect = element.getBoundingClientRect();
+
+    // Check that the element is in the viewport. This can be useful, for example, to check
+    // that an element is sticky.
+    const inViewport =
+      rect.top >= 0 &&
+      rect.left >= 0 &&
+      rect.bottom <=
+        (window.innerHeight || document.documentElement.clientHeight) &&
+      rect.right <= (window.innerWidth || document.documentElement.clientWidth);
+
+    // Also check that the element is not obscured by another element on top.
+    const centerX = rect.left + rect.width / 2;
+    const centerY = rect.top + rect.height / 2;
+    const topmostElement = document.elementFromPoint(centerX, centerY);
+
+    return inViewport && topmostElement == element;
+  }, selector);
+};
+
+// -----------------------------------------------------------------------
+// Snapshot tests
+// -----------------------------------------------------------------------
+
+test.describe("Qiskit top nav bar", () => {
+  test("does not cover Furo's top nav bar when scrolled down", async ({
+    page,
+  }) => {
+    await page.goto("sphinx_guide/lists.html");
+    await scrollDown(page, 200);
+
+    const pageToCVisible = await isVisibleInViewport(
+      page,
+      "div.toc-title-container"
+    );
+    expect(pageToCVisible).toBe(true);
+
+    const searchVisible = await isVisibleInViewport(
+      page,
+      "input.sidebar-search"
+    );
+    expect(searchVisible).toBe(true);
+
+    await setMobile(page);
+    const mobileHeaderVisible = await isVisibleInViewport(
+      page,
+      "header.mobile-header"
+    );
+    expect(mobileHeaderVisible).toBe(true);
+  });
+
+  test("does not cover the top of # anchor links", async ({ page }) => {
+    const checkHeader = async () => {
+      await page.goto("sphinx_guide/lists.html#definition-lists");
+      // We have to wait for the page animation to update.
+      await page.waitForTimeout(800);
+      const headerVisible = await isVisibleInViewport(
+        page,
+        "section#definition-lists > h2"
+      );
+      expect(headerVisible).toBe(true);
+    };
+
+    await checkHeader();
+
+    await setMobile(page);
+    // Go to a new page first to reset the scroll state.
+    await page.goto("");
+    await checkHeader();
+  });
+
+  test("does not cover the side menus when expanded on mobile", async ({
+    page,
+  }) => {
+    await setMobile(page);
+    await page.goto("");
+
+    const click = async (selector) => {
+      await page.locator(selector).click();
+      // Wait for the page to update.
+      await page.waitForTimeout(500);
+    };
+
+    await click("div.header-right label.toc-overlay-icon i");
+    const pageToCVisible = await isVisibleInViewport(
+      page,
+      "div.toc-title-container"
+    );
+    expect(pageToCVisible).toBe(true);
+
+    // Reload the page to close the nav bar.
+    await page.goto("");
```

Comment:
I tried to close it by clicking, but was getting issues about the thing not being clickable by Playwright etc. After 15 minutes, I gave up and went for the simple fix.

Clicking to close works as a human.